.gitignore

```diff
@@ -1 +1,7 @@
-target
+# ignore rust stuff
+Cargo.lock
```

Comment:
There is basically never a good reason not to track `Cargo.lock` in git. 

For bin crates it provides a known working set of dependencies. For lib crates it also makes sense to include the lockfile, even though it will be ignored by consumers of your lib, because it makes it easier to reproduce bugs that only happen with specific versions of dependencies (works in `cargo test` but does not work when used as a lib).

See: https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control

Reply:
ah interesting